### PR TITLE
Add unit tests for DebugEditorCtrl

### PR DESCRIPTION
**Pull request by sarahquigley**
This pull request adds unit tests for the DebugEditorCtrl

### CODE REVIEW

File: app/editor/editor.spec.coffee [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  describe 'DebugEditorCtrl', ->
+
+    DebugParser = undefined
+    $scope = undefined
+
+    beforeEach ->
+      # Create mock DebugParser
+      DebugParser = jasmine.createSpyObj('DebugParser', ['compile_graph'])
+      DebugParser.compile_graph.and.callFake((nodes) ->
+        return {nodes: nodes}
```
Comment (mstewart) [inline app/editor/editor.spec.coffee]: This should return a Graph instance, right?

File: app/editor/editor.spec.coffee [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+          DebugParser: DebugParser
+        })
+      )
+
+    it 'should define a $scope variable, $scope.nodes', ->
+      expect($scope.nodes).toEqual(jasmine.any(Object))
+
+    it 'should define a $scope variable, $scope.graph, to equal result of DebugParser.compile_graph of $scope.nodes', ->
+      expect(DebugParser.compile_graph).toHaveBeenCalledWith($scope.nodes)
+      expect($scope.graph).toEqual({nodes: $scope.nodes})
```
Comment (mstewart) [inline app/editor/editor.spec.coffee]: The graph returned could be extracted out into a constant to make these assertions a bit clearer.

File: app/editor/editor.spec.coffee [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+          expect($scope.nodes.test_node_2).toEqual('test2')
+
+    describe '.add_node', ->
+      describe 'if a node with the node_id of the new node does not exist', ->
+        it 'should add a new node', ->
+          expect($scope.nodes.test_node).toBeUndefined()
+          $scope.add_node('test_node', 'test')
+          expect($scope.nodes.test_node).toEqual('test')
+
+      describe 'if a node with the node_id of the new node does not exist', ->
```
Comment (mstewart) [inline app/editor/editor.spec.coffee]: s/does not exist/does exist/

Comment (sarahquigley) [inline app/editor/editor.spec.coffee]: Fixed.

**Comment by mstewart** [comment]
:sheep:

**Pull request merged**